torch/testing/_internal/common_nn.py

```diff
@@ -4938,31 +4984,32 @@ def map_tensor_sizes(sizes):
 
         return self._unpack(self._arg_cache[name]) if unpack else self._arg_cache[name]
 
-    def _get_input(self, unpack=True):
-        return self._get_arg('input', unpack)
-
-    def __call__(self, test_case):
-        raise NotImplementedError
+    def _get_input(self):
+        input = self._get_arg('input', False)  # type: ignore[arg-type]
 
+        def map_variables(i):
+            if isinstance(i, torch.Tensor):
+                if i.is_floating_point() or i.is_complex():
+                    i.requires_grad = True
+                return i
+            else:
+                return type(i)(map_variables(elem) for elem in i)
 
-class ModuleTest(TestBase):
+        return map_variables(input)
 
-    @abstractmethod
-    def _do_test(self, test_case: Any, module: nn.Module, input: Any) -> Any:
-        raise NotImplementedError
+    def __call__(self, test_case):
+        if self.is_criterion_test:
+            return self._criterion_test_call(test_case)
+        else:
+            return self._module_test_call(test_case)
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.jacobian_input = kwargs.get('jacobian_input', True)
-        self.should_test_cuda = kwargs.get('test_cuda', True)
-        self.should_test_pickle = kwargs.get('pickle', True)
-        self.check_gradgrad = kwargs.get('check_gradgrad', True)
-        self.FIXME_no_cuda_gradgrad_comparison = \
-            kwargs.get('FIXME_no_cuda_gradgrad_comparison', False)
-        self.precision = kwargs.get('precision', 2e-4)
-        self.check_forward_only = kwargs.get('check_forward_only', False)
+    def test_cuda(self, test_case, dtype=None, extra_args=None):
```

Comment:
Is this some legacy cruft that should be replaced by instantiating tests across multiple device types programmatically using our device generic test framework?